src/time-range-picker/time-range-picker.tsx

```diff
@@ -0,0 +1,220 @@
+            setShowEndTimeSelector(true);
+        }
+    }, [showEndTimeSelector]);
+
+    const handleMouseDownEvent = (event: MouseEvent) => {
+        if (!disabled) {
+            runOutsideFocusHandler(event);
+        }
+    };
+
+    const handleKeyUpEvent = (event: KeyboardEvent) => {
+        switch (event.code) {
+            case "Tab":
+                runOutsideFocusHandler(event);
+                break;
+            default:
+                break;
+        }
+    };
+
+    const handleSelectionDropdownCancel = () => {
+        setShowEndTimeSelector(false);
+        setShowStartTimeSelector(false);
+        onSelectionCancel && onSelectionCancel();
+    };
+
+    const handleStartTime = (value: string) => {
+        setShowStartTimeSelector(false);
+        setShowEndTimeSelector(true);
+        setStartTimeVal(value);
+
+        const timeValue: TimeRangePickerProps = {
+            startTime: startTimeVal,
+            endTime: endTimeVal,
+        };
+
+        onChange && onChange(timeValue);
+    };
+
+    const handleEndTime = (value: string) => {
+        setEndTimeVal(value);
+        if (startTimeVal == "") {
+            setShowStartTimeSelector(true);
+        }
+        if (startTimeVal > value) {
+            setEndTimeVal(startTimeVal);
+            handleStartTime(value);
+        }
+        setShowEndTimeSelector(false);
+
+        const timeValue: TimeRangePickerProps = {
+            startTime: startTimeVal,
+            endTime: endTimeVal,
+        };
+
+        onChange && onChange(timeValue);
+    };
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    const runOnBlurHandler = () => {
+        setShowStartTimeSelector(false);
+        setShowEndTimeSelector(false);
+        onBlur && onBlur();
+    };
+
+    const runOutsideFocusHandler = (event: MouseEvent | KeyboardEvent) => {
+        if (nodeRef && !nodeRef.current.contains(event.target as any)) {
+            if (!showEndTimeSelector) {
+                runOnBlurHandler();
+            }
+            if (!showStartTimeSelector) {
+                runOnBlurHandler();
+            }
+        }
+    };
+
+    return (
+        <Wrapper ref={nodeRef} id={id} {...otherProps}>
+            <TimeContainer
+                $disabled={disabled}
+                $error={error}
+                $readOnly={readOnly}
+            >
+                <InputSelectorStartTimeElement
+                    onFocus={handleStartTimeFocus}
+                    $focused={showStartTimeSelector}
+                    readOnly={readOnly}
+                    placeholder={"From"}
+                    value={TimeRangePickerHelper.formatValue(
+                        startTimeVal,
+                        format
+                    )}
+                    defaultValue={defaultValue}
+                    $disabled={disabled}
```

Comment:
inputs come with a `disabled` prop, you should use that instead to get the correct native browser behaviour.